source/game/field/RKGndCol.cc

```diff
@@ -0,0 +1,374 @@
+#include "RKGndCol.hh"
+
+#include <cmath>
+
+#include "egg/math/Math.hh"
+
+#include "game/field/KCollisionTypes.hh"
+
+// Some functions from m8 -
+// https://github.com/em-eight/mkw/tree/master/source/game/field/RKGndCol.cpp
+
+namespace Field {
+
+RKGndCol::RKGndCol() = default;
+
+RKGndCol::RKGndCol(const void *file) {
+    auto addOffset = [](const void *file, u32 offset) -> const void * {
+        return reinterpret_cast<const void *>(reinterpret_cast<const u8 *>(file) + offset);
+    };
+    u8 *unsafeData = reinterpret_cast<u8 *>(const_cast<void *>(file));
+    EGG::RamStream stream = EGG::RamStream(unsafeData, sizeof(KColHeader));
+
+    u32 posOffset = stream.read_u32();
+    u32 nrmOffset = stream.read_u32();
+    u32 prismOffset = stream.read_u32();
+    u32 blockOffset = stream.read_u32();
+
+    assert(posOffset == sizeof(KColHeader));
+
+    m_posData = addOffset(file, posOffset);
+    m_nrmData = addOffset(file, nrmOffset);
+    m_prismData = addOffset(file, prismOffset);
+    m_blockData = addOffset(file, blockOffset);
+
+    m_prismThickness = stream.read_f32();
+    m_areaMinPos.read(stream);
+    m_areaXWidthMask = stream.read_u32();
+    m_areaYWidthMask = stream.read_u32();
+    m_areaZWidthMask = stream.read_u32();
+    m_blockWidthShift = stream.read_u32();
+    m_areaXBlocksShift = stream.read_u32();
+    m_areaXYBlocksShift = stream.read_u32();
+    m_sphereRadius = stream.read_f32();
+
+    m_pos.setZero();
+    m_prevPos.setZero();
+    m_movement.setZero();
+    m_radius = 0.0f;
+    m_prismIter = nullptr;
+    m_cachedPrismArray = m_prismCache.data() - 1;
+    m_prisms = reinterpret_cast<const KCollisionPrism *>(
+            addOffset(m_prismData, sizeof(KCollisionPrism)));
+    m_prismCount =
+            (reinterpret_cast<uintptr_t>(m_blockData) - reinterpret_cast<uintptr_t>(m_prisms + 1)) /
+            sizeof(KCollisionPrism);
+
+    computeBBox();
+}
+
+void RKGndCol::computeBBox() {
+    m_bbox.mMax.set(-999999.0f);
+    m_bbox.mMin.set(999999.0f);
```

Comment:
It really seems like these are arbitrary values. Even so, we should be able to call `m_bbox.resetBound`

Reply:
Unfortunately, `resetBound` and these constants are different. They likely are arbitrary values.